src/builder/QuarkBuilder.sol

```diff
@@ -301,79 +294,49 @@ contract QuarkBuilder {
             plannedBridgeAmount += bridgeActionContext.amount;
         }
 
-        // Verify transfer actions is affordable
+        // Verify transfer actions are affordable
+        // NOTE: Assume all transfer actions are on the TransferIntent.chainId as Bridging logics is currently assuming destination at TransferIntent.chainId
+        // NOTE: To support multi-chain transfers, call below functions to check repeatedly with each chainId and plannedBridgeAmount
+        // for each chain (Likely passed from TransferIntent with a list of chain Id)
+        assertTransferActionsAffordableOnTargetChain(
+            transferActions, chainAccountsList, transferIntent.chainId, plannedBridgeAmount
+        );
+    }
+
+    function assertTransferActionsAffordableOnTargetChain(
+        Actions.Action[] memory transferActions,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        uint256 targetChainId,
+        uint256 plannedBridgeAmountToTargetChain
+    ) internal pure {
         uint256 accruedCost = 0;
```

Comment:
nit: `transferTokensUsed` is clearer

Reply:
🤔 
But cost is not necessary to be transfertokenused?

Reply:
oops, I meant `paymentTokensUsed`

Reply:
Not blocking, we can always refactor in a future PR

Reply:
done 👍